### PR TITLE
[flutter_tools] Add --build-number and --build-name support to web and linux

### PR DESCRIPTION
**Pull request by JankoLancer**
Added support to --build-number and --build-name build options for web and linux. Given values will be written in version.json.

### Related Issue
https://github.com/flutter/flutter/issues/75909

###Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 3 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/100377

File: packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart [hunk 1]
```diff
@@ -133,6 +135,9 @@ void main() {
     expect(output.childFile('kernel_blob.bin'), exists);
     expect(output.childFile('AssetManifest.json'), exists);
     expect(output.childFile('version.json'), exists);
+    final String versionFile = output.childFile('version.json').readAsStringSync();
+    expect(versionFile, contains('"version":"2.0.0"'));
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: why is the key here `version`?

Comment (JankoLancer) [reply packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: `version` key is used for representing `Project` version.
getVersionInfo method from Project is responsible for creating content for version.json and this method is using that key.
Maybe we can move these keys to some const values.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: does `kBuildName` somewhere get mapped to `version`? I'm looking at the diff and it looks like you're asserting that adding `"BuildName": "2.0.0"` to the defines result in `"version": "2.0.0"` being written to `version.json`, which is a little surprising to me.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: Note, if this is the case, it's not a problem with your change perse, but surprising.

Comment (JankoLancer) [reply packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: No, no. `kBuildName` is used in `BuildInfo` defines, for transfering data from build info to actual build.
You can see it here in action:
`    if (defines.containsKey(kBuildName)) {
      versionInfo['version'] = defines[kBuildName];
}`

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: So why not call this `build_name` in the version.json?

Comment (JankoLancer) [reply packages/flutter_tools/test/general.shard/build_system/targets/linux_test.dart:139]: That would require changes to packages that already use it as ```version. ``` I don't think we should change it.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
Merging with failing cirrus customer_testing as LUCI customer_testing is green.

**Pull request merged**